R/plugin-easybutton.R

```diff
@@ -0,0 +1,186 @@
+leafletEasyButtonDependencies <- function() {
+  list(
+    htmltools::htmlDependency(
+      "leaflet-easybutton",
+      "1.3.1",
+      system.file("htmlwidgets/plugins/Leaflet.EasyButton", package = "leaflet"),
+      script = c("easy-button.js", "EasyButton-binding.js"),
+      stylesheet = c('easy-button.css')
+    )
+  )
+}
+
+#' Create an easyButton statestate
+#' @param stateName a unique name for the state
+#' @seealso \code{\link{easyButton}}
+#' @describeIn easyButton state of an easyButton.
+#' @export
+easyButtonState <- function(
+   stateName,
+   icon,
+   title,
+   onClick
+) {
+  if(!inherits(onClick,'JS_EVAL')) {
+    stop("onClick needs to be a returned value from a JS() call")
+  }
+  structure(list(
+    stateName = as.character(stateName),
+    icon = as.character(icon),
+    title = as.character(title),
+    onClick = onClick
+  ),
+      class='leaflet_easybutton_state')
+}
+
+#' Creates an easy button.
+#' @seealso \url{https://github.com/CliffCloud/Leaflet.EasyButton}
+#' @param icon the button icon
+#' @param title text to show on hover
+#' @param onClick the action to take
+#' @param position topleft|topright|bottomleft|bottomright
+#' @param id id for the button
+#' @param states the states
+#' @export
+easyButton <- function(
```

Comment:
It looks like you're using parameter defaults as documentation. Can you not do that? The default values should be an actual reasonable default that the user would want, or nothing. In this case definitely the default `title` and `onClick`, at least, should be nothing--we want to force people to provide those.

(Only exception being a vector of possible choices, but even then, only if the first element is actually a reasonable default value.)